### ISSUE TITLE
v3: The yaml.Node struct cannot be marshaled

### ISSUE THREAD

**Issue by movsb** [go-yaml/yaml#673]
Example code:

```go
package main

import (
	"gopkg.in/yaml.v3"
)

func main() {
	var node yaml.Node
	yaml.Marshal(node)
}
```

Panics:

```-
$ go run main.go 
panic: reflect.Value.Addr of unaddressable value [recovered]
        panic: reflect.Value.Addr of unaddressable value

goroutine 1 [running]:
gopkg.in/yaml%2ev3.handleErr(0xc00013fed0)
        /Users/tao/code/go/pkg/mod/gopkg.in/yaml.v3@v3.0.0-20200615113413-eeeca48fe776/yaml.go:294 +0x9a
panic(0x10f47c0, 0x1140b10)
        /usr/local/Cellar/go/1.14.4/libexec/src/runtime/panic.go:969 +0x166
reflect.Value.Addr(0x1113d80, 0xc000136320, 0x99, 0x10cd701, 0x1113d80, 0xc000136320)
        /usr/local/Cellar/go/1.14.4/libexec/src/reflect/value.go:258 +0x96
gopkg.in/yaml%2ev3.(*encoder).marshal(0xc000158000, 0x0, 0x0, 0x1113d80, 0xc000136320, 0x99)
        /Users/tao/code/go/pkg/mod/gopkg.in/yaml.v3@v3.0.0-20200615113413-eeeca48fe776/encode.go:123 +0x837
gopkg.in/yaml%2ev3.(*encoder).marshalDoc(0xc000158000, 0x0, 0x0, 0x1113d80, 0xc000136320, 0x99)
        /Users/tao/code/go/pkg/mod/gopkg.in/yaml.v3@v3.0.0-20200615113413-eeeca48fe776/encode.go:105 +0x12f
gopkg.in/yaml%2ev3.Marshal(0x1113d80, 0xc000136320, 0x0, 0x0, 0x0, 0x0, 0x0)
        /Users/tao/code/go/pkg/mod/gopkg.in/yaml.v3@v3.0.0-20200615113413-eeeca48fe776/yaml.go:222 +0x32b
main.main()
        /Users/tao/code/test/main.go:9 +0x89
exit status 2
```

https://github.com/go-yaml/yaml/blob/eeeca48fe7764f320e4870d231902bf9c1be2c08/encode.go#L123

**Comment by niemeyer** [comment]
Try providing a pointer to it instead.

**Comment by movsb** [comment]
> Try providing a pointer to it instead.

@niemeyer Not a good idea, because other structs are ok.

**Comment by niemeyer** [comment]
@movsb You're right, sorry. I've fixed this in 749611f.
